core/trino-parser/src/main/java/io/trino/sql/parser/AstBuilder.java

```diff
@@ -470,6 +474,21 @@ public Node visitSetSchemaAuthorization(SqlBaseParser.SetSchemaAuthorizationCont
                 getPrincipalSpecification(context.principal()));
     }
 
+    private static SaveMode toSaveMode(TerminalNode replace, TerminalNode exists)
+    {
+        boolean isReplace = replace != null;
+        boolean isNotExists = exists != null;
```

Comment:
Add a checkArgument to verify that isReplace and isNotExists are not simultaneously true. That would indicate there's a bug in the caller.